ext/spl/spl_fixedarray.c

```diff
@@ -93,7 +93,7 @@ static void spl_fixedarray_init(spl_fixedarray *array, zend_long size) /* {{{ */
 
 static void spl_fixedarray_resize(spl_fixedarray *array, zend_long size) /* {{{ */
 {
-	if (size == array->size) {
+	if (size == array->size || (size == 0 && array->elements == NULL)) {
```

Comment:
Why do we have to explicitly check this?

Reply:
We need to prevent recursive resizing to 0, but this is indeed better be done in the `size == 0` branch below.

Reply:
Ah, I see. I think it would be better to also move the array->size = 0 before the dtors then.